src/qat/model/serialisation.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+import typing
+from typing import Annotated, Dict, List
+
+from pydantic import BeforeValidator, PlainSerializer
+
+from qat.model.component import Component, ComponentId
+
+T = typing.TypeVar("T", bound=Component)
+
+SerializeItemToRefs = PlainSerializer(
+    lambda o: o.model_dump_id(), return_type=dict, when_used="always"
+)
+Ref = Annotated[ComponentId | T, SerializeItemToRefs, "Ref"]
+
+SerializeDictToRefs = PlainSerializer(
+    lambda d: {k.uuid: v.model_dump_id() for (k, v) in d.items()},
+    return_type=dict,
+    when_used="always",
+)
+
+
+def deserializeDictToRefs(data):
```

Comment:
Super minor point but isn't the standard in python to have function names with underscores instead of caps? e.g. `deserialize_dict_to_refs(data)`. I personally don't mind as long as it is consistent :)

Also typo below "Reydrates"